offer/src/main/java/org/shareio/backend/core/usecases/service/SearchOffersUseCaseService.java

```diff
@@ -23,13 +27,59 @@ public class SearchOffersUseCaseService implements SearchOffersUseCaseInterface
 
     GetUserProfileDaoInterface getUserProfileDaoInterface;
     GetAllOffersDaoInterface getAllOffersDaoInterface;
+    GetLocationUseCaseInterface getLocationUseCaseInterface;
     GetAllUserIdListUseCaseInterface getAllUserIdListUseCaseInterface;
     GetAverageUserReviewValueUseCaseInterface getAverageUserReviewValueUseCaseInterface;
 
     @Override
     public List<UUID> getOfferListMeetingCriteria(UUID userId, String title, String category, String condition, Double distance, Double score, LocalDate creationDate, String sortType) {
-        User user = getUserProfileDaoInterface.getUserDto(userId).map(User::fromDto).orElseThrow(NoSuchElementException::new);
         List<Offer> offers = getAllOffersDaoInterface.getAllOffers().stream().map(Offer::fromDto).toList();
+        User user = getUserProfileDaoInterface.getUserDto(userId).map(User::fromDto).orElseThrow(NoSuchElementException::new);
```

Comment:
User was pulled before offers, because, if he doesn't exist, we don't want to waste time getting all offers